### PR TITLE
Integrate kubernetes security into console client code

### PR DESCRIPTION
**Pull request by rmohr**
In order to leverege kubernetes security mechanisms, we need to wrap our
websocket connections with the kubernetes http round trippers.

### CODE REVIEW

File: pkg/virtctl/console/console.go [hunk 1]
```diff
@@ -58,37 +60,50 @@ func (c *Console) Run(flags *flag.FlagSet) int {
...
+
+func WebsocketCallback(ws *websocket.Conn, resp *http.Response, err error) error {
 
-	ws, resp, err := websocket.DefaultDialer.Dial(u.String(), nil)
 	if err != nil {
 		if resp != nil && resp.StatusCode != http.StatusOK {
 			buf := new(bytes.Buffer)
 			buf.ReadFrom(resp.Body)
-			log.Fatalf("Can't connect to console (%d): %s\n", resp.StatusCode, buf.String())
+			return fmt.Errorf("Can't connect to console (%d): %s\n", resp.StatusCode, buf.String())
```
Comment (stu-gott) [inline pkg/virtctl/console/console.go]: Kuberenetes convention is to always use lower case error messages (because they generally get chained together). Is that a convention we follow?

File: pkg/virtctl/console/console.go [hunk 2]
```diff
@@ -135,11 +150,78 @@ func (c *Console) Run(flags *flag.FlagSet) int {
...
+		return nil, err
+	}
+
+	switch u.Scheme {
+	case "https":
+		u.Scheme = "wss"
+	case "http":
+		u.Scheme = "ws"
+	default:
+		return nil, fmt.Errorf("Malformed URL %s", u.String())
```
Comment (stu-gott) [inline pkg/virtctl/console/console.go]: Is there a way to explain "we only support http and https" without getting too wordy? It might help users understand what's wrong with the URL.

**Comment by stu-gott** [comment]
LGTM

File: pkg/virtctl/console/console.go (hunk 1 above)
Comment (rmohr) [inline pkg/virtctl/console/console.go]: Are they chaining them with semicolons?
